maven-java/kalix-maven-plugin/src/main/java/kalix/DeployMojo.java

```diff
@@ -36,30 +36,102 @@ public class DeployMojo extends AbstractMojo {
     @Parameter(property = "kalixContext")
     private String kalixContext;
 
+    @Parameter(property = "kalixCurrentProject", required = true)
+    private String kalixCurrentProject;
+
+    @Parameter(defaultValue = "false", property = "deployToKalix", required = true)
+    private Boolean deployToKalix;
+
     @Parameter(defaultValue = "30000", property = "cliTimeoutMs", required = true)
     private Long cliTimeoutMs;
 
     private final Log log = getLog();
 
     /**
-     * We deploy by invoking the services deploy command
+     * We deploy by invoking the services deploy command only when `deployToKalix` is `true`
+     * and the current Kalix project matches with 'kalixCurrentProject'.
      */
     public void execute() throws MojoExecutionException {
-        log.info("Deploying project to Kalix");
+       final List<String> commandLine;
+       final int deploymentResult = 0;
+       if (!Boolean.valueOf(deployToKalix)){
+           log.info("The variable `deployToKalix` hasn't been set to `true`. Therefore, not deploying to Kalix");
+           log.info("To deploy to Kalix via `mvn deploy` you need to set `deployToKalix` to `true` " +
+                   "and `kalixCurrentProject` to your current Kalix project Name (`kalix config get-project`)");
```

Comment:
I think this behaviour wrt the cli project selection is weird.

I think it should just be `kalixProject` and then we deploy using `kalix --project [project]`, so that a mvn project defines what project to deploy it to regardless of cli project selection.

I think that would also make it nicely possible to deploy the same project to different projects, from ci for example, with just mvn sysprops, without having to first switch kalix current project.

Reply:
Or did I imagine that the cli takes such a flag?

Reply:
It does accept --project but there is no mention in the help screen.

Reply:
@johanandren, do you mean that we should not check if the current selected project matches the project in the maven config?

Reply:
For clarity, what is not clear to me is if you saying that we should only have `kalixProject` property and remove the boolean `deployToKalix`? 

So, the behavior would be: always deploy to the project that we explicitly said to deploy. And eventually, using a env var to change to some else.